doc/user_manual/advanced_users_plugins.tex

```diff
@@ -0,0 +1,347 @@
+call it at the initialization stage of each ``Step'' (see section \ref{sec:steps}).
+%
+RAVEN will communicate, thorough a set of method attributes, all the information
+that are generally needed to perform an initialization:
+\begin{itemize}
+  \item runInfo, a dictionary containing information regarding how the
+  calculation is set up (e.g. number of processors, etc.).
+  %
+  It contains the following attributes:
+  \begin{itemize}
+    \item \texttt{DefaultInputFile} -- default input file to use
+    \item \texttt{SimulationFiles} -- the xml input file
+    \item \texttt{ScriptDir} -- the location of the pbs script interfaces
+    \item \texttt{FrameworkDir} -- the directory where the framework is located
+    \item \texttt{WorkingDir} -- the directory where the framework should be
+    running
+    \item \texttt{TempWorkingDir} -- the temporary directory where a simulation
+    step is run
+    \item \texttt{NumMPI} -- the number of mpi process by run
+    \item \texttt{NumThreads} -- number of threads by run
+    \item \texttt{numProcByRun} -- total number of core used by one run (number
+    of threads by number of mpi)
+    \item \texttt{batchSize} -- number of contemporaneous runs
+    \item \texttt{ParallelCommand} -- the command that should be used to submit
+    jobs in parallel (mpi)
+    \item \texttt{numNode} -- number of nodes
+    \item \texttt{procByNode} -- number of processors by node
+    \item \texttt{totalNumCoresUsed} -- total number of cores used by driver
+    \item \texttt{queueingSoftware} -- queueing software name
+    \item \texttt{stepName} -- the name of the step currently running
+    \item \texttt{precommand} -- added to the front of the command that is run
+    \item \texttt{postcommand} -- added after the command that is run
+    \item \texttt{delSucLogFiles} -- if a simulation (code run) has not failed,
+    delete the relative log file (if True)
+    \item \texttt{deleteOutExtension} -- if a simulation (code run) has not
+    failed, delete the relative output files with the listed extension (comma
+    separated list, for example: `e,r,txt')
+    \item \texttt{mode} -- running mode, curently the only mode supported is
+      mpi (but custom modes can be created)
+    \item \textit{expectedTime} -- how long the complete input is expected to
+    run
+    \item \textit{logfileBuffer} -- logfile buffer size in bytes
+  \end{itemize}
+  \item inputs, a list of all the inputs that have been specified in the
+  ``Step'' using this model.
+  %
+\end{itemize}
+As all the others method in the ExternalModel Plugin, the information \emph{must} be
+stored in the ``self-like'' object ``container''.
+In the following an example is reported:
+\begin{lstlisting}[language=python]
+def initialize(self, container, runInfo, inputs):
+ # Let's suppose we just need to initialize some variables
+  container.sigma = 10.0
+  container.rho   = 28.0
+  # end function
+\end{lstlisting}
+
```

Comment:
I did not included it yet since Probably we would like to follow the idea to have a documentation for plugins. A documentation for code interfaces. etc.
(I will open an issue for that)